lib/inventories/permission.rb

```diff
@@ -0,0 +1,107 @@
+module Inventories
+  class Permission
+    PERMISSIONS = [:facilitator, :participant]
+
+    attr_reader :inventory
+    attr_reader :user
+
+    def initialize(inventory:, user:)
+      @inventory = inventory
+      @user = user
+    end
+
+    def possible_roles_permissions
+      PERMISSIONS-[get_level(user)]
+    end
+
+    def requested
+      AccessRequest.where(assessment_id: assessment.id)
+    end
+
+    def get_access_request(user)
+      AccessRequest.find_by(assessment_id: assessment.id, user_id: user.id)
```

Comment:
This model - `AccessRequest` - may be a good candidate to become polymorphic to increase its reusability.